src/mapper_class.cpp

```diff
@@ -80,34 +79,20 @@ void MapperClass::Initialize(ros::NodeHandle *nh) {
     // Load radius for radius collision-checking
     nh->getParam("radius_collision_check", radius_collision_check_);
 
-    // Load depth camera names
-    std::vector<std::string> depth_cam_names;
-    std::string depth_cam_prefix, depth_cam_suffix;
-    nh->getParam("depth_cam_names", depth_cam_names);
-    nh->getParam("depth_cam_prefix", depth_cam_prefix);
-    nh->getParam("depth_cam_suffix", depth_cam_suffix);
-
-    // Load lidar topic names
-    std::vector<std::string> lidar_names;
-    std::string lidar_prefix, lidar_suffix;
-    nh->getParam("lidar_names", lidar_names);
-    nh->getParam("lidar_prefix", lidar_prefix);
-    nh->getParam("lidar_suffix", lidar_suffix);
+    // Load lidar and base_link pose topics
+    std::string lidar_topic, base_link_pose_topic;
+    nh->getParam("lidar_topic", lidar_topic);
+    nh->getParam("base_link_pose_topic", base_link_pose_topic);
 
     // Load frame ids
-    std::vector<std::string> cam_frame_id;
-    std::vector<std::string> lidar_frame_id;
+    std::string lidar_frame_id, robot_base_link_frame_id;
     nh->getParam("inertial_frame_id", inertial_frame_id_);
-    nh->getParam("robot_frame_id", robot_frame_id_);
-    nh->getParam("cam_frame_id", cam_frame_id);
+    nh->getParam("robot_base_link_frame_id", robot_base_link_frame_id);
     nh->getParam("lidar_frame_id", lidar_frame_id);
 
-    // Check if number of cameras/lidar added match the number of frame_id for each of them
-    if (depth_cam_names.size() != cam_frame_id.size()) {
-        ROS_ERROR("[mapper]: Number of cameras is different from camera tf frame_ids!");
-    }
-    if (lidar_names.size() != lidar_frame_id.size()) {
-        ROS_ERROR("[mapper]: Number of lidar topics is different from lidar tf frame_ids!");
+    // Load transform between base_link and lidar
+    if (!helper::LookupTransform(robot_base_link_frame_id, lidar_frame_id, &tf_lidar_in_base_link_frame_)) {
+        return;
```

Comment:
If there's not an error log here, I would definitely make the LookupTransform method throw an error, since this is a critical lookup